assets/wizards/newspack/views/settings/connections/custom-events.tsx

```diff
@@ -0,0 +1,143 @@
+/**
+ * Settings Wizard: Connections > Custom Events
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { WizardError } from '../../../../errors';
+import { useWizardApiFetch } from '../../../../hooks/use-wizard-api-fetch';
+import { Button, Grid, Notice, TextControl } from '../../../../../components/src';
+
+/**
+ * Validate GA4 Measurement ID.
+ *
+ * @see   https://measureschool.com/ga4-measurement-id/
+ *
+ * @param measurementId Measurement ID to validate
+ * @return boolean      True if the measurement ID is valid, false otherwise
+ */
+function isValidGA4MeasurementID( measurementId = '' ) {
+	const ga4Pattern = /^G-[A-Za-z0-9]{10,}$/;
+	return ga4Pattern.test( measurementId );
+}
+
+/**
+ * Analytics Custom Events screen.
+ */
+function CustomEvents() {
+	const [ ga4Credentials, setGa4Credentials ] = useState< Ga4Credentials >(
+		window.newspackSettings.connections.sections.customEvents
+	);
+	const { wizardApiFetch, errorMessage, resetError, setError } = useWizardApiFetch(
+		'newspack-settings/connections/custom-events'
+	);
+
+	useEffect( () => {
+		if ( ! isValidGA4MeasurementID( ga4Credentials.measurement_id ) ) {
```

Comment:
This results in an error always being shown even when a publisher does not want to enable custom events. This feels a bit strange. At the very least we should not show the error message when both custom event fields are empty. But IMO an error message should only appear in response to some invalid action taken by the user (i.e. perform some validation before saving and outputting the error at this stage).

Reply:
Thanks for the feedback @chickenn00dle, I have added an early exit to avoid errors displaying when both inputs are empty.

Reply:
Great! This is much better 👍